src/Sulu/Bundle/AdminBundle/Resources/js/components/DatePicker/tests/DatePicker.test.js

```diff
@@ -0,0 +1,69 @@
+/* eslint-disable flowtype/require-valid-file-annotation */
+import React from 'react';
+import {render, mount} from 'enzyme';
+import DatePicker from '../DatePicker';
+
+beforeEach(() => {
+    const constantDate = new Date('2017-05-13 06:32:20');
+    Date.now = jest.genMockFunction().mockReturnValue(constantDate);
+});
+
+test('DatePicker should render', () => {
+    const onChange = jest.fn();
+    expect(render(<DatePicker onChange={onChange} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render with placeholder', () => {
+    const onChange = jest.fn();
+    expect(render(<DatePicker placeholder="My placeholder" onChange={onChange} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render with value', () => {
+    const onChange = jest.fn();
+    expect(render(<DatePicker value="2017-05-05" onChange={onChange} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render null value as empty string', () => {
+    const onChange = jest.fn();
+    expect(render(<DatePicker value={null} onChange={onChange} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render date format only with month', () => {
+    const onChange = jest.fn();
+    const options = {
+        dateFormat: 'MMMM',
+    };
+    expect(render(<DatePicker value={null} onChange={onChange} options={options} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render date format only with year', () => {
+    const onChange = jest.fn();
+    const options = {
+        dateFormat: 'YYYY',
+    };
+    expect(render(<DatePicker value={null} onChange={onChange} options={options} />)).toMatchSnapshot();
+});
+
+test('DatePicker should render date picker with time picker', () => {
+    const onChange = jest.fn();
+    const options = {
+        timeFormat: true,
+    };
+    expect(render(<DatePicker value={null} onChange={onChange} options={options} />)).toMatchSnapshot();
+});
+
+test('DatePicker should set class correctly when overlay was opened/closed', () => {
+    const onChange = jest.fn();
+    const input = mount(<DatePicker value={null} onChange={onChange} />);
+
+    // overlay should be closed
+    expect(input.find('.rdt').hasClass('rdtOpen')).toBe(false);
+
+    // open dialog and check if class is set
+    input.find('Input Icon span').simulate('click');
+    expect(input.find('.rdt').hasClass('rdtOpen')).toBe(true);
+
+    // choose a date and check if class was removed again
+    input.find('.rdtPicker tbody tr td').first().simulate('click');
+    expect(input.find('.rdt').hasClass('rdtOpen')).toBe(false);
```

Comment:
I would also check here if the date has been written to the input, I think that is not tested yet.

Reply:
It is tested now